components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -886,6 +886,94 @@ public API getAPI(APIIdentifier identifier,APIIdentifier oldIdentifier, String o
     /**
      * Returns a list of pre-defined # {@link org.wso2.carbon.apimgt.api.model.Tier} in the system.
      *
+     * @return Set<Tier>
+     */
+    public Set<Tier> getAppTiers() throws APIManagementException {
+        Set<Tier> tiers = new TreeSet<Tier>(new TierNameComparator());
+
+        Map<String, Tier> tierMap;
+        if (tenantId == 0) {
+            tierMap = APIUtil.getAppTiers();
+        } else {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId, true);
+            tierMap = APIUtil.getAppTiers(tenantId);
+            PrivilegedCarbonContext.endTenantFlow();
```

Comment:
If there is a exception throws, the tenant flow will not be end. May be you can close it in finally block. However it seems this code is no longer available :)